src/components/Card.js

```diff
@@ -1,20 +1,31 @@
 class Card {
-  constructor({ name, link }, cardSelector, cardImageClickHandler) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    cardImageClickHandler,
+    handleDelete,
+    cardLikeClick
+  ) {
+    this._isLiked = isLiked;
+    this._handleCardLikeClick = cardLikeClick;
     this._name = name;
     this._link = link;
+    this.id = _id;
     this._cardSelector = cardSelector;
     this._cardImageClickHandler = cardImageClickHandler;
+    this._handleDelete = handleDelete;
   }
 
   _setEventListeners() {
-    const likeButton = this._cardElement.querySelector(".card__like-button");
-    likeButton.addEventListener("click", () => {
-      likeButton.classList.toggle("card__like-button_active");
+    this._likeButton.addEventListener("click", () => {
+      this._isLiked = !this._isLiked;
+      this.updateLikeState();
+      this._handleCardLikeClick(this.id, this._isLiked);
     });
     const removeButton = this._cardElement.querySelector(".delete__button");
     removeButton.addEventListener("click", () => {
       const cardItem = removeButton.closest(".card");
-      cardItem.remove();
+      this._handleDelete(this, cardItem);
     });
 
     const cardImageElement = this._cardElement.querySelector(".card__image");
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```